### PR TITLE
Refactorization of getFacilityByPerunBeans

### PR DESCRIPTION
**Pull request by metodej**
- method was split into several overloaded methods

### CODE REVIEW

File: perun-core/src/main/java/cz/metacentrum/perun/core/bl/FacilitiesManagerBl.java [hunk 1]
```diff
@@ -782,16 +781,64 @@
...
-	 * Get perunBean and try to find all connected Facilities
+	 * @param sess
+	 * @param group
+	 * @return list of facilities connected with group
+	 * @throws InternalErrorException
+	 */
+	List<Facility> getFacilitiesByPerunBean(PerunSession sess, Group group) throws InternalErrorException;
+
+	/**
+	 * Returns list of facilities with a member
```
Comment (stavamichal) [inline perun-core/src/main/java/cz/metacentrum/perun/core/bl/FacilitiesManagerBl.java]: Connected with a member. Or "by member".

File: perun-core/src/main/java/cz/metacentrum/perun/core/bl/FacilitiesManagerBl.java [hunk 2]
```diff
@@ -782,16 +781,64 @@
...
+	 *
+	 * @param sess
+	 * @param member
+	 * @return list of facilities connected with member
+	 * @throws InternalErrorException
+	 */
+	List<Facility> getFacilitiesByPerunBean(PerunSession sess, Member member) throws InternalErrorException;
+
+	/**
+	 * Returns list of facilities with a resource
```
Comment (stavamichal) [inline perun-core/src/main/java/cz/metacentrum/perun/core/bl/FacilitiesManagerBl.java]: Connected with a resource. Or "by resource".

File: perun-core/src/main/java/cz/metacentrum/perun/core/bl/FacilitiesManagerBl.java [hunk 3]
```diff
@@ -782,16 +781,64 @@
...
+	 *
+	 * @param sess
+	 * @param resource
+	 * @return list of facilities connected with resource
+	 * @throws InternalErrorException
+	 */
+	List<Facility> getFacilitiesByPerunBean(PerunSession sess, Resource resource) throws InternalErrorException;
+
+	/**
+	 * Returns list of facilities with a user
```
Comment (stavamichal) [inline perun-core/src/main/java/cz/metacentrum/perun/core/bl/FacilitiesManagerBl.java]: Connected with user. Or "by user".

File: perun-core/src/main/java/cz/metacentrum/perun/core/bl/FacilitiesManagerBl.java [hunk 4]
```diff
@@ -782,16 +781,64 @@
...
+	 *
+	 * @param sess
+	 * @param user
+	 * @return list of facilities connected with user
+	 * @throws InternalErrorException
+	 */
+	List<Facility> getFacilitiesByPerunBean(PerunSession sess, User user) throws InternalErrorException;
+
+	/**
+	 * Returns list of facilities with a host
```
Comment (stavamichal) [inline perun-core/src/main/java/cz/metacentrum/perun/core/bl/FacilitiesManagerBl.java]: Connected with host. Or "by host".

File: perun-core/src/main/java/cz/metacentrum/perun/core/bl/FacilitiesManagerBl.java [hunk 5]
```diff
@@ -782,16 +781,64 @@
...
+	 *
+	 * @param sess
+	 * @param host
+	 * @return list of facilities connected with host
+	 * @throws InternalErrorException
+	 */
+	List<Facility> getFacilitiesByPerunBean(PerunSession sess, Host host) throws InternalErrorException;
+
+	/**
+	 * Returns list of facilities with a vo
```
Comment (stavamichal) [inline perun-core/src/main/java/cz/metacentrum/perun/core/bl/FacilitiesManagerBl.java]: Connected with vo. Or "by vo".

File: perun-core/src/main/java/cz/metacentrum/perun/core/bl/FacilitiesManagerBl.java (hunk 1 above)
Comment (metodej) [reply perun-core/src/main/java/cz/metacentrum/perun/core/bl/FacilitiesManagerBl.java]: I overlooked it, thanks, it should be fixed now.

**Pull request merged**